xilem/examples/http_cats.rs

```diff
@@ -111,28 +94,28 @@ impl HttpCats {
             ))
             .must_fill_major_axis(true),
             worker(
-                worker_value,
                 |proxy, mut rx| async move {
-                    while let Some(request) = rx.recv().await {
-                        if let Some(code) = request {
-                            let proxy = proxy.clone();
-                            tokio::task::spawn(async move {
-                                let url = format!("https://http.cat/{code}");
-                                let result = image_from_url(&url).await;
-                                match result {
-                                    // We choose not to handle the case where the event loop has ended
-                                    Ok(image) => drop(proxy.message((code, image))),
-                                    // TODO: Report in the frontend
-                                    Err(err) => {
-                                        tracing::warn!(
-                                            "Loading image for HTTP status code {code} from {url} failed: {err:?}"
-                                        );
-                                    }
+                    while let Some(code) = rx.recv().await {
+                        let proxy = proxy.clone();
+                        tokio::task::spawn(async move {
+                            let url = format!("https://http.cat/{code}");
+                            let result = image_from_url(&url).await;
+                            match result {
+                                // We choose not to handle the case where the event loop has ended
+                                Ok(image) => drop(proxy.message((code, image))),
+                                // TODO: Report in the frontend
+                                Err(err) => {
+                                    tracing::warn!(
+                                        "Loading image for HTTP status code {code} from {url} failed: {err:?}"
+                                    );
                                 }
-                            });
-                        }
+                            }
+                        });
                     }
                 },
+                |state: &mut Self, sender| {
+                    state.download_sender = Some(sender);
```

Comment:
To be clear, I meant having this callback be of the form `state.download_receiver.take().expect("We only ever create one worker")`. 
An argument in terms of this being a different kind of mutation doesn't seem coherent to me, but maybe I'm misunderstanding.